src/Blorc.OpenIdConnect/Services/AccessTokenExpirationDelegatingHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Blorc.OpenIdConnect;
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class AccessTokenExpirationDelegatingHandler : DelegatingHandler
+{
+    private readonly IUserManager _userManager;
+
+    private readonly TimeProvider _timeProvider;
+
+    public AccessTokenExpirationDelegatingHandler(IUserManager userManager, TimeProvider timeProvider)
+    {
+        ArgumentNullException.ThrowIfNull(userManager);
+        ArgumentNullException.ThrowIfNull(timeProvider);
+
+        _userManager = userManager;
+        _timeProvider = timeProvider;
+    }
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        ArgumentNullException.ThrowIfNull(request);
+
+        var user = await _userManager.GetUserAsync<User<Profile>>();
+        if (user?.ExpiresAt is not null)
+        {
+            var expiryDateTime = DateTimeOffset.FromUnixTimeSeconds(user.ExpiresAt).UtcDateTime;
+            if (_timeProvider.GetUtcNow() > expiryDateTime)
+            {
+                await _userManager.SignOutRedirectAsync();
```

Comment:
should we return here? or continue processing?

Reply:
Good catch! I had the same doubt, but since the return will never be executed (redirect), So, I think it's 'fine' in this way.

Reply:
Let me try something